### PR TITLE
test: added tests for reporters

### PR DESCRIPTION
**Pull request by Demivan**
Additional changes:

- Json reporter now used Date.now() for start time to match Jest
- Prettify json output (add newlines and indentation)
@cexbrayat Are these changes fine?

### CODE REVIEW

**Review by cexbrayat — APPROVED** [review]
LGTM 👍 Good job with the tests

**Comment by Demivan** [comment]
@patak-dev PR ready for review

File: test/reporters/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "compilerOptions": {
+        "types": ["vitest/global"],
```
Comment (patak-dev) [inline test/reporters/tsconfig.json:3]:
```suggestion
        "types": ["vitest/globals"],
```

**Comment by patak-dev** [comment]
Awesome work @Demivan 👏🏼

**Review by patak-dev — APPROVED** [review]

**Pull request merged**